packages/database/src/core/operation/Operation.ts

```diff
@@ -42,40 +41,38 @@ export interface Operation {
   operationForChild(childName: string): Operation | null;
 }
 
-export class OperationSource {
-  constructor(
-    public fromUser: boolean,
-    public fromServer: boolean,
-    public queryId: string | null,
-    public tagged: boolean
-  ) {
-    assert(!tagged || fromServer, 'Tagged queries must be from server.');
-  }
+export interface OperationSource {
+  fromUser: boolean;
+  fromServer: boolean;
+  queryId: string | null;
+  tagged: boolean;
+}
 
-  static user() {
-    return new OperationSource(
-      /*fromUser=*/ true,
-      false,
-      null,
-      /*tagged=*/ false
-    );
-  }
+export function newOperationSourceUser(): OperationSource {
```

Comment:
Why are these all prefixed with "new"?

Reply:
They are factory functions. I can remove the prefix if you think we shouldn't do this for consistency - let me know,

Reply:
That makes sense.